pkg/logging/logging.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2021 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package fybriklogging
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+)
+
+// LogType indicates the types of log entries supported.  Logging packages typically provide separate function
+// calls for the different log types.
+type LogType string
+
+const (
+	ERROR LogType = "error" // Errors encountered in the code
+	INFO  LogType = "info"  // Info logs are things you want to tell the user or administrator which are not errors.
+)
+
+// VerbosityLevel allows one to decide how "chatty" you want your logs to be.
+type VerbosityLevel int
+
+const (
+	MANDATORY VerbosityLevel = 0
+	DEBUG     VerbosityLevel = 1
+	TRACE     VerbosityLevel = 2
+)
+
+// LogEntry defines the information for a single entry stored in the system logs.
+// It is assumed that the timestamp is added automatically and thus not included in the log entry structure.
+// Verbosity and log type are added by the log library via which the entries are sent to stdout or stderr.
+type LogEntry struct {
+	// Caller indicates the cluster, component and function that generated the message in the format "cluster/component/function"
+	// Use the buildCallerPath function to construct this.
+	// +required
+	Caller string `json:"caller"`
+
+	// Message being logged.
+	// +required
+	Message string `json:"message"`
+
+	// Unique id of the kubernetes fybrikapplication (Needs to be unique over time in order to support history and not just current workloads)
+	// +required
+	FybrikAppGUID string `json:"fybrikAppGuid"`
+
+	// ForUser should be true if the message should be shared with the end user in the FybrikApplication status.
+	// Assumed false if not indicated.
+	// +optional
+	ForUser bool `json:"forUser,omitempty"`
```

Comment:
Not sure I understand. If the message should be shared with the end user in the status, manager will update the status. It is not as if manager will collect the logs, analyze them and update the status when ForUser is true.

Reply:
@shlomitk1 This is not meant to replace the status returned in fybrikapplication.status.  It duplicates it, so that if a customer deploys logging tools like logDNA or fluentd, etc they can filter and act on things at the log level.  See the different uses of the logs from the perspective of the different actors as defined in https://docs.google.com/document/d/1x3Jnp3CIN6YhJKtWM1uIimaw9-q6Hb9M7LqkazmBRBo/edit#heading=h.z3iyvac259h6

Reply:
@simanadler Do you have an example when ForUser is set to true but ForArchive/Audit isn't? In the wrapper functions either both are true or both are false, unless I missed something.

Reply:
@shlomitk1 When we add support for reconciling based on policy, metadata or infrastructure updates then we will have situations where Audit (ForArchive) will be true but ForUser will be false.